File: Dataset/Leetcode/train/69/403.c
int XXX(int x){
    // 二分法
    int l = 0;
    int r = x;
    int k;

    while (l <= r) {
        int m = l + (r - l) / 2;
        long long v = (long long)m * m;
        if (v == x) return m; // 如果恰好等于就是答案
        else if (v < x) { // 否则记录最后一次比 x 小的 m，是最终答案
            k = m;
            l = m + 1;
        } else {
            r = m - 1;
        }
    }
    return k;
}

File: Dataset/Leetcode/train/22/403.py
 class Solution:
    def XXX(self, n: int) -> List[str]:
        res = []
        def valid(arr):
            res = []
            for i in range(len(arr)):
                countL = 0
                countR = 0
                for s in arr[i]:
                    if s == '(':
                        countL += 1
                    else:
                        countR += 1
                    if countR>countL:
                        break
                if countR == countL:
                    s = ''.join(arr[i])
                    if s not in res: 
                        res.append(''.join(arr[i])) 
            return res
        def dfs(paths,num,lnum,rnum):
            if len(paths)==num*2:
                res.append(paths)
                return
            if lnum<num:
                dfs(paths+'(',num,lnum+1,rnum)
            if rnum<num:
                dfs(paths+')',num,lnum,rnum+1)
        dfs('',n,0,0)
        return valid(res)


